### DISCUSSION TITLE
Having trouble setting up Azure OpenAI

### DISCUSSION THREAD

**Discussion by marius-smg** [FlowiseAI/Flowise#483] [Q&A]
Hey i'm struggeling with setting up Azure OpenAi for a document Q&A Chain. I have tried several different  things for the azure openai fields but always get error Codes (400, 404, 401). Most recently i get Error Code 401.

Has anyone experience in setting up Azure OpenAi instead of OpenAI and can help me?

In my Azure Portal i have both deployed text-embedding-ada-002 and gpt-35-turbo

![Bildschirmfoto 2023-07-05 um 16 35 24](https://github.com/FlowiseAI/Flowise/assets/132438025/c5999395-18a1-4b9d-a86f-52327fda1dd3)

**Comment by Soumyasds** [comment]
I face the same issue on my flow for Azure openAI

**Reply by marius-smg** [reply]
Have you made any progress? I somehow managed to get past the 400, 404, and 401 errors but now i get 500 erorrs :D

**Reply by Soumyasds** [reply]
My flow is working fine now. GPT-35 turbo did not work at all in my case. I used to get 401 errors and 500 internal server error when internal-prediction API was called.  

So, I used davinci-003 and it worked for me without any issues. Make sure you have a deployment with this model on your azure open AI

**Reply by marius-smg** [reply]
hm i only can deploy gpt-35-turbo and text-embedding-ada-002. davinci-003 isnt listed in my azure portal. i got a simple chatflow working with only gpt-35-turbo but i need a document Q&A setup so the text-embedding-ada-002 is needed for the embeddings.

**Reply by marius-smg** [reply]
ok i got it working: i put in text-embedding-ada-002 but i should have put in the name i gave this deployment
